overlord/assertstate/assertstate.go

```diff
@@ -880,7 +880,7 @@ func TryEnforcedValidationSets(st *state.State, validationSets []string, userID
 // of validation set keys to validation sets, pinned sequence numbers (if any),
 // installed snaps and ignored snaps. It fetches any pre-requisites necessary.
 func ApplyEnforcedValidationSets(st *state.State, valsets map[string]*asserts.ValidationSet, pinnedSeqs map[string]int, snaps []*snapasserts.InstalledSnap, ignoreValidation map[string]bool, userID int) error {
```

Comment:
this can potentially talk to the store which is something we don't want during seeding, what do we do with validation-sets assertion in the seed? the issue is around account and account-keys needed for them

Reply:
Ah right, this also explains the `devicePastSeeding` requirement. Okay, so what we need is a variant of that function that accomplishes the same without fetching/contacting the store. I added this